### ISSUE TITLE
PrivateMessages.UnreadUpdated event is not being fired under certain conditions

### ISSUE THREAD

**Issue by nmur** [sirkris/Reddit.NET#105]
I created a new project today with the first goal being to just monitor unread messages, but wasn't able to get it to work until I compared it directly to the `RedditTests.ControllerTests.WorkflowTests.PrivateMessagesTests.Conversation` test. The difference that I ended up finding is as below:

```
using Reddit;
using Reddit.Controllers;
using Reddit.Controllers.EventArgs;
using System;

namespace ConsoleApp1
{
    class Program
    {
        static void Main(string[] args)
        {
            var reddit = new RedditClient(/*credentials*/);

            FailingMonitorMessages(reddit);
            //SucceedingMonitorMessages(reddit.Account.Messages);
        }

        private static void FailingMonitorMessages(RedditClient reddit)
        {
            reddit.Account.Messages.MonitorUnread();
            reddit.Account.Messages.UnreadUpdated += C_UnreadMessagesUpdated;

            DateTime start = DateTime.Now;
            while (start.AddMilliseconds(60000) > DateTime.Now) { }

            reddit.Account.Messages.UnreadUpdated -= C_UnreadMessagesUpdated;
            reddit.Account.Messages.MonitorUnread();
        }

        private static void SucceedingMonitorMessages(PrivateMessages messages)
        {
            messages.MonitorUnread();
            messages.UnreadUpdated += C_UnreadMessagesUpdated;

            DateTime start = DateTime.Now;
            while (start.AddMilliseconds(60000) > DateTime.Now) { }

            messages.UnreadUpdated -= C_UnreadMessagesUpdated;
            messages.MonitorUnread();
        }

        private static void C_UnreadMessagesUpdated(object sender, MessagesUpdateEventArgs e)
        {
            Console.WriteLine("Message!");
        }
    }
}
```

There are two methods for monitoring messages. If you use an account with an unread message, and then run this application, the first method, which is uncommented, fails to fire the event handler when new unread messages are present. If you comment out the first method and use the second method instead, it works fine. 
Notice that the difference between the two is only that with the successful method, we're passing in the reference to `reddit.Account.Messages` instead of just the reference to `reddit`. This was the difference between the test and my project ([as well as the ELIZA sample code](https://github.com/sirkris/Reddit.NET/blob/master/docs/examples/cs/src/ELIZA.cs#L43), which I couldn't get to work either). 

I don't actually know why it works one way and not the other, maybe I'm overlooking or lacking some fundamental C# behaviour, or maybe it's a bug.

**Comment by GamingWolf** [comment]
It also counts already read pms as unread on startup and modmails just permanently fires.

**Comment by kyleratti** [comment]
@GamingWolf If you happen to be still searching for a solution on this, see #117

**Comment by sirkris** [comment]
I just pushed a fix to #117.  Could you let me know if it also fixes your issue?  You can find it on the devleop branch.

**Comment by sirkris** [comment]
Scratch that, it doesn't fix the issue.  The problem was a bug in the Account Controller.  The Messages property's get accessor wasn't actually storing the value when initialized, which apparently caused the event not to fire properly.  It's fixed now.  Nice catch!

Merged into 1.5.
